stellargraph/datasets/datasets.py

```diff
@@ -304,3 +305,108 @@ class AIFB(
+):
+    def load(self):
+        """
+        Load this dataset into a list of StellarGraph objects with corresponding labels, downloading it if required.
+
+        Note: Edges in MUTAG are labelled as one of 4 values: aromatic, single, double, and triple indicated by integers
+        0, 1, 2, 3 respectively. The edge labels are included in the  :class:`StellarGraph` objects as edge weights in
+        integer representation.
+
+        Returns:
+            A tuple that is a list of :class:`StellarGraph` objects and a Pandas Series of labels one for each graph.
+        """
+        self.download()
+        return self._load_from_location(location=self.data_directory)
+
+    def _load_from_txt_file(
+        self, location, filename, names=None, dtype=None, index_increment=None
+    ):
+        df = pd.read_csv(
+            os.path.join(location, filename),
+            header=None,
+            index_col=False,
+            dtype=dtype,
+            names=names,
+        )
+        if index_increment:
+            df.index = df.index + index_increment
+        return df
+
+    def _load_from_location(self, location):
+
+        if not os.path.isdir(location):
+            raise NotADirectoryError(
+                "The location {} is not a directory.".format(location)
+            )
+
+        df_graph = self._load_from_txt_file(
+            location=location, filename="MUTAG_A.txt", names=["source", "target"]
+        )
+        df_edge_labels = self._load_from_txt_file(
+            location=location,
+            filename="MUTAG_edge_labels.txt",
+            names=["weight"],
+            dtype=int,
+        )
+        df_graph = pd.concat([df_graph, df_edge_labels], axis=1)  # add edge weights
+        df_graph_ids = self._load_from_txt_file(
+            location=location,
+            filename="MUTAG_graph_indicator.txt",
+            names=["graph_id"],
+            index_increment=1,
+        )
+        df_graph_labels = self._load_from_txt_file(
+            location=location,
+            filename="MUTAG_graph_labels.txt",
+            dtype="category",
+            names=["label"],
+            index_increment=1,
+        )  # binary labels {-1, 1}
+        df_node_labels = self._load_from_txt_file(
+            location=location,
+            filename="MUTAG_node_labels.txt",
+            dtype="category",
+            index_increment=1,
+        )
+
+        # Let us one-hot encode the node labels because these are used as node features
+        # in graph classification tasks.
+        df_node_labels = pd.get_dummies(df_node_labels)
+
+        graphs = []
+        for graph_id in np.unique(df_graph_ids):
+            # find the subgraph with nodes that correspond to graph_id
+            node_ids = list(
+                df_graph_ids.loc[df_graph_ids["graph_id"] == graph_id].index
+            )
+
+            df_subgraph = df_graph[
+                df_graph["source"].isin(node_ids) | df_graph["target"].isin(node_ids)
```

Comment:
It probably doesn't matter, but strictly, this should be "and", not "or", because otherwise a subgraph could end up with dangling edges pointing to unknown nodes (i.e. not in `node_ids`)

```suggestion
                df_graph["source"].isin(node_ids) & df_graph["target"].isin(node_ids)
```

However, I assume that the dataset is designed so that there's no such edges, and the source and target of an edge always have the same graph ID; as in, the graph ID of the source uniquely determines which graph it is in. As such, this could be simplified to:

```suggestion
                df_graph["source"].isin(node_ids)
```

(This suggestion applies to `graph_for_nodes` in the group-by comment above.)

Reply:
Yes, for this dataset, either of the 3 methods works the same.